src/lib/checkbox/checkbox.scss

```diff
@@ -293,6 +300,13 @@ $_mat-checkbox-mark-stroke-size: 2 / 15 * $mat-checkbox-size !default;
         right: $_mat-checkbox-item-spacing;
       }
     }
+
+    &.mat-checkbox-inner-container-no-side-margin {
```

Comment:
You don't need to define this class twice; just one without the parent part of the selector would be better.